src/plugins/expressions/common/execution/execution_contract.test.ts

```diff
@@ -59,7 +59,7 @@ describe('ExecutionContract', () => {
 
   test('can cancel execution', () => {
     const execution = createExecution('foo bar=123');
-    const spy = jest.spyOn(execution, 'cancel');
+    const spy = jest.spyOn(execution, 'cancel').mockImplementation(() => {});
```

Comment:
I noticed that hits were silently failing tests (on master)
```
(node:57027) UnhandledPromiseRejectionWarning: AbortError: Aborted
(node:57027) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:57027) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I assume that it warns is that `cancel` is called without start and there is no one subscribed to handled inner AbortRejection promise of the execution.
I mocked it, since this unit test for execution_contract and we are not testing an execution here.

Reply:
But really, probably we should also handle and test edge case: 

```
const ex = new Execution();
ex.cancel() // ?? what happens? 
ex.start() // ?? what happens? 
```